.travis.yml

```diff
@@ -16,12 +21,34 @@ before_install:
   - yes | sdkmanager "platforms;android-30"
 
 script:
-  - cd plugin
+  # front-facing api unit tests
+  - cd api
+
+  # download Gut (Godot unit testing) library
+  - wget -O Gut.zip $GUT_URL
+  - unzip Gut.zip
+  - cp -a Gut-${GUT_VERSION}/addons/ addons/
+
+  # download Godot headless
+  - wget -O ${GODOT_HEADLESS}.zip $GODOT_HEADLESS_URL
+  - unzip ${GODOT_HEADLESS}.zip
+  - chmod +x ${GODOT_HEADLESS}
+  
+  # run test cases
+  - ./${GODOT_HEADLESS} --path $PWD -s addons/gut/gut_cmdln.gd -d -gexit -ginclude_subdirs -gdir=res://test
+
+
+  # plugin unit tests
+  - cd ../plugin
```

Comment:
nit: Rather than cd-ing into `../plugin`, you might be able to use `$TRAVIS_BUILD_DIR/plugin` (found [here](https://stackoverflow.com/questions/14006810/how-do-i-get-travis-build-working-directory-in-travis-yml)). That way you don't have to rely on the current directory being one level below $TRAVIS_BUILD_DIR

Reply:
Sounds good! This also make the configuration file less dependent to the existing project structure, making it easier if we want to change the project structure.